src/transformers/models/tvp/modeling_tvp.py

```diff
@@ -0,0 +1,914 @@
+        return pixel_values
+
+
+# Pad frames extracted from videos in the surroundings.
+class TvpFramePadPrompter(nn.Module):
+    def __init__(self, image_size, prompt_size, frame_num, max_img_size, visual_prompter_apply):
+        if visual_prompter_apply not in ("add", "replace", "remove"):
+            raise ValueError("`visual_prompter_apply` must be in (add, replace, remove)")
+
+        super(TvpFramePadPrompter, self).__init__()
+        visual_prompt_size = prompt_size
+        image_size = image_size
+        self.frame_num = frame_num
+        self.max_img_size = max_img_size
+        self.visual_prompter_apply = visual_prompter_apply
+
+        self.base_size = image_size - visual_prompt_size * 2
+        self.pad_up = nn.Parameter(torch.randn([1, frame_num, 3, visual_prompt_size, image_size]))
+        self.pad_down = nn.Parameter(torch.randn([1, frame_num, 3, visual_prompt_size, image_size]))
+        self.pad_left = nn.Parameter(
+            torch.randn([1, frame_num, 3, image_size - visual_prompt_size * 2, visual_prompt_size])
+        )
+        self.pad_right = nn.Parameter(
+            torch.randn([1, frame_num, 3, image_size - visual_prompt_size * 2, visual_prompt_size])
+        )
+
+    def forward(self, pixel_values):
+        if self.visual_prompter_apply != "add":
+            visual_prompt_mask = torch.ones([self.max_img_size, self.max_img_size], dtype=pixel_values.dtype)
+            pixel_values *= visual_prompt_mask
+        if self.visual_prompter_apply != "remove":
+            base = torch.zeros(1, self.frame_num, 3, self.base_size, self.base_size)
+            prompt = torch.cat([self.pad_left, base, self.pad_right], dim=4)
+            prompt = torch.cat([self.pad_up, prompt, self.pad_down], dim=3)
+            prompt = torch.cat(pixel_values.size(0) * [prompt])
+            pixel_values += prompt.to(pixel_values.dtype)
+        return pixel_values
+
+
+TVP_PROMPTER_CLASSES_MAPPING = {
+    "framedownpad": TvpFrameDownPadPrompter,
+    "framepad": TvpFramePadPrompter,
+}
+
+
+@add_start_docstrings(
+    "The bare Tvp Model transformer outputting BaseModelOutputWithPooling object without any specific head on" " top.",
+    TVP_START_DOCSTRING,
+)
+class TvpModel(TvpPreTrainedModel):
+    def __init__(self, config):
+        super(TvpModel, self).__init__(config)
+        self.config = config
+        self.vision_model = TvpVisionModel(config)
+        self.transformer = TvpTransformer(config)
+        if config.visual_prompter_type not in TVP_PROMPTER_CLASSES_MAPPING:
+            raise ValueError("`visual_prompter_type` must be in (framedownpad, framepad)")
+        else:
+            self.visual_prompter = TVP_PROMPTER_CLASSES_MAPPING[config.visual_prompter_type](
+                config.max_img_size,
+                config.visual_prompt_size,
+                config.num_frames,
+                config.max_img_size,
+                config.visual_prompter_apply,
+            )
```

Comment:
These layers should just take the `config` as their input argument and use that to instantiate themselves